src/main/scala/org/deeplearning4j/scalphagozero/board/GameState.scala

```diff
@@ -22,12 +19,23 @@ class GameState(
     val lastMove: Option[Move]
 ) {
 
-  var allPreviousStates: Set[(Player, Long)] = previousState match {
-    case None => Set[(Player, Long)]()
-    case Some(state) =>
-      state.allPreviousStates += ((nextPlayer, state.board.zobristHash))
-      state.allPreviousStates
-  }
+  private val allPreviousStates: Set[(Player, Long)] =
+    previousState match {
+      case None        => Set.empty
+      case Some(state) => state.allPreviousStates ++ Set(nextPlayer -> state.board.zobristHash)
+    }
```

Comment:
Here I'm not sure that my modification keep the old behavior.

In the old behavior the `previousState.allPreviousStates` content was modified while in my version it's not.

For me, it's strange to modify `previousState.allPreviousStates` in the way it was modified.

Is this a bug or the normal behavior ?